vendor/parser-extended/dune

```diff
@@ -3,7 +3,12 @@
  (public_name ocamlformat-lib.parser_extended)
  (flags
   (:standard -w -9 -open Parser_shims -open Ocaml_common))
- (libraries compiler-libs.common menhirLib parser_shims ocaml_common))
+ (libraries
+  erase_jane_syntax
```

Comment:
Yes, that's right -- I didn't think to look closely at passing it in like that, since I was concerned about Menhir entry points, but I'll take a look, possibly make the change, and document things either way.

Reply:
I'm inclined to leave it as is: I need to set something up that's available to Menhir and then set globally, and this same thing also has to be available in `Ast_helper` or I have to propagate booleans there, and given all that (plus the merge benefits), there doesn't seem to be much of a real locality win?